### PR TITLE
Introduction to GitHub -- Sami Darrow

### PR DESCRIPTION
**Pull request by sbdarrow**
A blog post that highlights what GitHub is, what it's used for, and some of the highlights about the website.

### CODE REVIEW

File: _posts/2020-11-09-introduction-to-github.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title:  "Introduction to GitHub!"
+date:   2020-11-09 12:54:04 -0700
+categories: GitHub Introduction Basics
+---
+# Introduction to GithHub
+
+![GitHub Logo](https://upload.wikimedia.org/wikipedia/commons/thumb/9/95/Font_Awesome_5_brands_github.svg/124px-Font_Awesome_5_brands_github.svg.png)
```
Comment (sauerbach-code) [inline _posts/2020-11-09-introduction-to-github.md:10]: Good use of the logo!

File: _posts/2020-11-09-introduction-to-github.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+![version control]({{ site.url }}/assets/version_control.png)
+
+It is very important that you keep track of what you’ve done and what works when writing code. Version control goes hand-in-hand with collaborative coding. As previously mentioned, GitHub is great for collaborative projects because of its version control ability. If Person 2 were to work directly on the code that Person 1 was developing, that would break the code and ruin the whole project.
+
+GitHub allows for version control in several ways, some of which are detailed below. For continuity’s sake, I will be referencing the two-person project I detailed in the previous section:
+  * **The ability to fork a repository:** Person 2 can go to person 1’s GitHub profile and find the repository that has their project in it. They can then fork this repository to their GitHub account.
+    * This basically means that they copy the project to their GitHub account so they can use it as their own and not mess with what Person 1 is doing.
+
+
```
Comment (sauerbach-code) [inline _posts/2020-11-09-introduction-to-github.md:52]: Insert explanation of branches in this section here, it makes sense to talk about a branch you push to and pull from Github before mentioning pull requests, as this is the typical workflow with GitHub. You can also mention the ability to switch between branches on the command line.

File: _posts/2020-11-09-introduction-to-github.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  * **Pushing Code to GitHub:** if you are working on the command line this can be done through the command ‘git push’ followed by the name of the file that you would like to push to GitHub. This allows Person 2 to take whatever they were working on and put it onto GitHub for Person 1 to see.
+    * This step is not inherently about version control, but it is an important step that needs to be taken to maintain project continuity.
+
+
+  * **Creating a pull request:** Person 2 can create a pull request that will notify Person 1 that they made some changes to the existing code. When looking at the work Person 2 submitted, Person 1 can decide whether it will be helpful or whether it will harm the existing code.
+    * If they determine that it would harm the existing code, they would not merge it to the main repository. This is where version control is important because without it, the whole project could have been messed up.
+
```
Comment (sauerbach-code) [inline _posts/2020-11-09-introduction-to-github.md:63]: Make a note of what to do with a branch after a merge - whether to delete it or not.

File: _posts/2020-11-09-introduction-to-github.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+GitHub allows for version control in several ways, some of which are detailed below. For continuity’s sake, I will be referencing the two-person project I detailed in the previous section:
+  * **The ability to fork a repository:** Person 2 can go to person 1’s GitHub profile and find the repository that has their project in it. They can then fork this repository to their GitHub account.
+    * This basically means that they copy the project to their GitHub account so they can use it as their own and not mess with what Person 1 is doing.
+
+
+  * **Pushing Code to GitHub:** if you are working on the command line this can be done through the command ‘git push’ followed by the name of the file that you would like to push to GitHub. This allows Person 2 to take whatever they were working on and put it onto GitHub for Person 1 to see.
+    * This step is not inherently about version control, but it is an important step that needs to be taken to maintain project continuity.
+
```
Comment (sauerbach-code) [inline _posts/2020-11-09-introduction-to-github.md:55]: Make note about 'git pull' as well. You have to pull stuff from GitHub once a new version gets added, or if you have to start working on a new repository.

File: _posts/2020-11-09-introduction-to-github.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+date:   2020-11-09 12:54:04 -0700
+categories: GitHub Introduction Basics
+---
+# Introduction to GithHub
+
+![GitHub Logo](https://upload.wikimedia.org/wikipedia/commons/thumb/9/95/Font_Awesome_5_brands_github.svg/124px-Font_Awesome_5_brands_github.svg.png)
+
+Hello! In this blog post I will be going through a brief introduction to GitHub. I’ll be touching on what GitHub is, what it’s used for, and how useful it can be to anyone that codes! Hopefully if you are reading this you find it helpful!
+
+## What is GitHub
```
Comment (sauerbach-code) [inline _posts/2020-11-09-introduction-to-github.md:14]: Nice use of headers throughout the post!

File: _posts/2020-11-09-introduction-to-github.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+## The Importance of the Ability to Collaborate
+
+![collaborate](https://glip.com/content/dam/glip/us/en/images/pages/use_cases/development/development-content.svg)
+
+When working on a project, especially if you are working in a team, sharing the work you have done with each other is important. GitHub allows software developers to share their code and each person who is working on the project can do their own work without messing up the other persons progress.
+
+Say there are two people working on a project: Person 1 is the main coder who develops the base line code for everyone to work off of. Person 2 is a contributor that can also work on the code and provide feedback and maybe make some slight changes to what Person 1 has done. The basic workflow that would go into this two-person project would be as follows:
+  * **Person 1:** Creates working code for what the project calls for and pushes this code to GitHub
```
Comment (sauerbach-code) [inline _posts/2020-11-09-introduction-to-github.md:31]: Nice, succinct explanation!

**Review by sauerbach-code — CHANGES_REQUESTED** [review]
Hi Sami, 
This blogpost is generally pretty good, with nice organization and good explanations! There are a couple of things that need to be explained or elaborated on: mainly branches (like a few sentences explaining what they are, main branch, etc.) and using 'git pull' as well as 'git push'. Once those changes are made, this blogpost should be good to go!

File: _posts/2020-11-09-introduction-to-github.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  * **The ability to fork a repository:** Person 2 can go to person 1’s GitHub profile and find the repository that has their project in it. They can then fork this repository to their GitHub account.
+    * This basically means that they copy the project to their GitHub account so they can use it as their own and not mess with what Person 1 is doing.
+    * Additionally, you have the ability to create new branches in a project. This acts as an effective form of version control as well. A new branch essentially a copy of the original code but it will not effect any of the code on the main branch.
+      * Some command when working with branches are as follows:
+        * Creating a new branch: `git checkout -b <name-of-new-branch>`
+        * Switching to an existing branch: `git checkout <name-of-existing-branch>`
+        * Preview all existing branches: `git branch`
+
+
+  * **Pushing Code to GitHub:** if you are working on the command line this can be done through the command ‘git push’ followed by the name of the file that you would like to push to GitHub. This allows Person 2 to take whatever they were working on and put it onto GitHub for Person 1 to see.
```
Comment (naupaka) [inline _posts/2020-11-09-introduction-to-github.md]: Not precisely correct. It should be `git push` followed by the name of the remote (often `origin`) and then the name of the branch on the remote to push to (`master` or `main`). Git pushes and pulls commits, not files.

**Review by naupaka — CHANGES_REQUESTED** [review]
Great job on this. One very minor technical change that you should make before this is ready to merge.

**Comment by naupaka** [comment]
@sbdarrow please make the fix I suggested so that we can get this merged and get you the point

**Comment by sbdarrow** [comment]
Hi @naupaka just made the requested changes!

**Comment by naupaka** [comment]
Thanks! 👍

**Review by naupaka — APPROVED** [review]

**Comment by naupaka** [comment]
https://2020f-usfca-btec-640.github.io/github/introduction/basics/2020/11/09/introduction-to-github.html

**Pull request merged**